sale_stock_picking_note/models/stock_picking.py

```diff
@@ -1,10 +1,23 @@
 # Copyright 2021 Tecnativa - Víctor Martínez
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class StockPicking(models.Model):
     _inherit = "stock.picking"
 
     customer_note = fields.Text(string="Customer Comments")
+
+    @api.onchange("partner_id", "picking_type_id")
+    def _onchange_partner_id_picking_notes(self):
+        for picking in self:
+            if picking.picking_type_id.code != "incoming":
+                if self.group_id.sale_id:
+                    note = self.group_id.sale_id.picking_note
+                    customer_note = self.group_id.sale_id.picking_customer_note
+                else:
+                    note = picking.partner_id.picking_note
+                    customer_note = picking.partner_id.picking_customer_note
```

Comment:
suggestion:
```suggestion
                self.note = self.sale_id.picking_note or self.partner_id.picking_note
                self.customer_note = self.sale_id.picking_customer_note or self.partner_id.picking_customer_note
```

question: should we keep manual modifications from being squashed by this onchange?

Reply:
I think it's not necessary since the picking notes are related directly to the partner.
Maybe we could check if it's empty adding self.note and self.customer_note at the end of "or" chains.

```python
self.note = self.sale_id.picking_note or self.partner_id.picking_note or self.note
self.customer_note = self.sale_id.picking_customer_note or self.partner_id.picking_customer_note or self.customer_note
```

Reply:
I think we should always propagate the data from the order and the partner. 
I don't see the Stock User adding notes on the stock picking directly by himself. Maybe a Salesman receiving a call and adding it to an order, but not for the Stock User.

```python
# I prefer a depends to allow batch update of the sales
@api.depends("sale_id.picking_note", "partner_id.picking_note", "sale_id.picking_customer_note", "partner_id.picking_customer_note")
```

Reply:
Done both